spark/src/main/scala/ai/chronon/spark/batch/BatchNodeRunner.scala

```
@@ -160,36 +160,51 @@ class BatchNodeRunner(node: Node, tableUtils: TableUtils) extends NodeRunner {
+            (dayPartitionMillis)
+          }
 
-        logger.info(
-          s"Successfully persisted data quality metrics and schema mapping for table: $outputTable (${tileSummaries.size} tile summaries)")
-      } else {
-        logger.info(s"No tile summaries found for table: $outputTable")
+          val statsPutRequests = groupedTileSummaries.map { case ((dayPartitionMillis), columnTileSummaries) =>
+            val nullCountsStats = IcebergPartitionStatsExtractor.createNullCountsStats(columnTileSummaries)
+            val partitionStats = TileStats.nullCounts(nullCountsStats)
+            IcebergPartitionStatsExtractor.createPartitionStatsPutRequest(outputTable,
+                                                                          partitionStats,
+                                                                          dayPartitionMillis,
+                                                                          TileStatsType.NULL_COUNTS)
+          }.toSeq
+
+          statsExtractor.extractSchemaMapping(outputTable) match {
+            case Some(schemaMapping) =>
+              val schemaPutRequest =
+                IcebergPartitionStatsExtractor.createSchemaMappingPutRequest(outputTable, schemaMapping)
+              val allPutRequests = statsPutRequests :+ schemaPutRequest
+
+              try {
+                val kvStoreUpdates = metricsKvStore.multiPut(allPutRequests)
+                Await.result(kvStoreUpdates, 30.seconds)
+
+                logger.info(
+                  s"Successfully persisted data quality metrics and schema mapping for table: $outputTable (${tileSummaries.size} tile summaries)")
+              } catch {
+                case e: Exception =>
+                  logger.info(
+                    s"Failed to persist data quality metrics to KV store for table: $outputTable. This may be expected if the KV store table does not exist. Error: ${e.getMessage}")
```

might be useful to log the full exception (with the stack) as it might be due to a different error like a timeout / auth error which just the message might swallow